File: fix_contratos.php
<?php
$file = 'c:/Users/gesto/Desktop/projects/integra/app/Http/Controllers/ContratosController.php';
$content = file_get_contents($file);

// Replacement for NIT cleaning block
$oldBlock = "/\/\/ Limpiar NIT de formatos Excel \(decimales\) o guiones de verificación\s+\$nit_str = trim\(\(string\)\$nit_raw\);\s+if \(strpos\(\$nit_str, '-'|\"\-\"\) !== false\) \{\s+\$nit_parts = explode\('-'|\"\-\"|\'-\', \$nit_str\);\s+\$nit = trim\(\$nit_parts\[0\]\);\s+\} else \{\s+\$nit = preg_replace\('\/\.\\\.0\+\\\$\/'|\"\/\.\\\.0\+\\\$\/\", '', \$nit_str\);\s+\}/";

// Actually, let's just use string replace for a very specific part if regex is too complex
$newContent = $content;

// 1. First loop NIT cleaning
$target1 = '$nit_str = trim((string)$nit_raw);
            if (strpos($nit_str, \'-\') !== false) {
                $nit_parts = explode(\'-\', $nit_str);
                $nit = trim($nit_parts[0]);
            } else {
                $nit = preg_replace(\'/\.0+$/\', \'\', $nit_str);
            }';

// Since the spaces/tabs might be the issue, I'll use a regex for the white spaces
$pattern = '/\$nit_str = trim\(\(string\)\$nit_raw\);\s+if \(strpos\(\$nit_str, \'-\'\) !== false\) \{\s+\$nit_parts = explode\(\'-\', \$nit_str\);\s+\$nit = trim\(\$nit_parts\[0\]\);\s+\} else \{\s+\$nit = preg_replace\(\'\/\.0\+\\\$\/\', \'\', \$nit_str\);\s+\}/';
$replacement = '$nit = $this->cleanIdentification($nit_raw);';

$newContent = preg_replace($pattern, $replacement, $newContent);

if ($newContent !== null && $newContent !== $content) {
    file_put_contents($file, $newContent);
    echo "Successfully updated ContratosController.php\n";
} else {
    echo "Failed to update ContratosController.php or no changes needed\n";
}

File: fix_contratos.py
import re

path = r'c:\Users\gesto\Desktop\projects\integra\app\Http\Controllers\ContratosController.php'
with open(path, 'r', encoding='utf-8', errors='ignore') as f:
    content = f.read()

# Pattern for the NIT block
pattern = r'\$nit_str = trim\(\(string\)\$nit_raw\);\s+if \(strpos\(\$nit_str, [\'"]-[\'"]\) !== false\) \{\s+\$nit_parts = explode\([\'"]-[\'"], \$nit_str\);\s+\$nit = trim\(\$nit_parts\[0\]\);\s+\} else \{\s+\$nit = preg_replace\([\'"]\/\.0\+\\\$/[\'"], [\'"][\'"], \$nit_str\);\s+\}'
# Note: I'll make the regex even more flexible by using \s* and .*?
flexible_pattern = r'\$nit_str = trim\(\(string\)\$nit_raw\);\s*if \(strpos\(\$nit_str, [\'"]-[\'"]\) !== false\) \{.*?\$nit = preg_replace\([\'"]\/\.0\+\\?\$/[\'"], [\'"][\'"], \$nit_str\);\s*\}'

new_content = re.sub(flexible_pattern, '$nit = $this->cleanIdentification($nit_raw);', content, flags=re.DOTALL)

if new_content != content:
    with open(path, 'w', encoding='utf-8') as f:
        f.write(new_content)
    print("Successfully updated ContratosController.php")
else:
    print("Failed to update ContratosController.php")
